inference-engine/tests/functional/plugin/shared/src/single_layer_tests/gather_nd.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "ngraph_functions/builders.hpp"
+#include "single_layer_tests/gather_nd.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string GatherNDLayerTest::getTestCaseName(const testing::TestParamInfo<GatherNDParams>& obj) {
+    InferenceEngine::SizeVector dataShape, indicesShape;
+    InferenceEngine::Precision dPrecision, iPrecision;
+    int batchDims;
+    std::string device;
+    GatherNDParamsSubset gatherArgsSubset;
+    std::tie(gatherArgsSubset, dPrecision, iPrecision, device) = obj.param;
+    std::tie(dataShape, indicesShape, batchDims) = gatherArgsSubset;
+
+    std::ostringstream result;
+    result << "DS=" << CommonTestUtils::vec2str(dataShape) << "_";
+    result << "IS=" << CommonTestUtils::vec2str(indicesShape) << "_";
+    result << "BD=" << batchDims << "_";
+    result << "DP=" << dPrecision.name() << "_";
+    result << "IP=" << iPrecision.name() << "_";
+    result << "device=" << device;
+    return result.str();
+}
+
+void GatherNDLayerTest::SetUp() {
+    InferenceEngine::SizeVector dataShape, indicesShape;
+    InferenceEngine::Precision dPrecision, iPrecision;
+    int batchDims;
+    GatherNDParamsSubset gatherArgsSubset;
+    std::tie(gatherArgsSubset, dPrecision, iPrecision, targetDevice) = this->GetParam();
+    std::tie(dataShape, indicesShape, batchDims) = gatherArgsSubset;
+
+    auto ngDPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(dPrecision);
+    auto ngIPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(iPrecision);
+    auto ngParamType = ngDPrc;
+    if (ngDPrc == ngraph::element::i64 || ngDPrc == ngraph::element::i16) {
+        ngParamType = ngraph::element::f32;
+    }
+
+    auto params = ngraph::builder::makeParams(ngParamType, {dataShape});
+    auto paramOuts = ngraph::helpers::convert2OutputVector(
+            ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+    auto dataNode = paramOuts[0];
+    if (ngDPrc == ngraph::element::i64 || ngDPrc == ngraph::element::i16) {
```

Comment:
I believe will be enough to add comment in ```dPrecisions``` that we don't support I64 at this moment and comment out when it will be possible
anyway we convert I64 -> I32

Reply:
The same reason as above comment.